### PR TITLE
Add parameter to Maps API calls to identify client

### PR DESCRIPTION
**Pull request by jgoizueta**
See #444

This adds a `client=vl-VERSION` parameter to Maps API calls (both map instantiation and mvt tile fetches).

Should we send it only for map instantiation?

Other decisions I've taken:
* single parameter to identify client (CARTO VL) and version. The alternative would be to use separate `client`and `client-version` parameters
* short identification of client: `vl` not `carto-vl`
* version not prefixed by `v`

### CODE REVIEW

File: src/client/windshaft.js [hunk 1]
```diff
@@ -684,6 +685,7 @@ function authURL(url, conf) {
     if (conf.apiKey) {
         const sep = url.includes('?') ? '&' : '?';
         url += sep + 'api_key=' + encodeURIComponent(conf.apiKey);
+        url += `&client=vl-${version}`;
```
Comment (davidmanzanares) [inline src/client/windshaft.js]: encodeURIComponent() ?

File: src/client/windshaft.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import * as LRU from 'lru-cache';
 import * as windshaftFiltering from './windshaft-filtering';
 import { VectorTile } from '@mapbox/vector-tile';
 import Metadata from '../core/metadata';
+import { version } from '../../package.json';
```
[the remark this answers is not in the document]
Comment (jgoizueta) [reply src/client/windshaft.js]: I hope only the version, but I'm not a JS expert (I have no clue, really)

[the remark this answers is not in the document]
Comment (IagoLast) [reply src/client/windshaft.js]:
```js

/***/ }),
/* 39 */
/***/ (function(module, exports) {

module.exports = {"name":"@carto/carto-vl","version":"0.3.0","description":"CARTO Vector library","repository":{"type":"git","url":"git://github.com/CartoDB/carto-vl.git"},"author":{"name":"CARTO","url":"https://carto.com/"},"contributors":["David Manzanares <dmanzanares@carto.com>","Iago Lastra <iago@carto.com>","Jesús Arroyo Torrens <jarroyo@carto.com>","Javier Goizueta <jgoizueta@carto.com>","Mamata Akella <makella@carto.com>","Raúl Ochoa <rochoa@carto.com>","Ariana Escobar <ariana@carto.com>"],"license":"BSD-3-Clause","files":["src","dist"],"dependencies":{"@mapbox/vector-tile":"^1.3.0","cartocolor":"^4.0.0","earcut":"^2.1.2","jsep":"CartoDB/jsep#additional-char-ids-packaged","lru-cache":"^4.1.1","mitt":"^1.1.3","pbf":"^3.1.0"},"devDependencies":{"@carto/mapbox-gl":"0.45.0-carto1","chai":"^4.1.2","chai-as-promised":"^7.1.1","eslint":"^4.15.0","exquisite-sst":"IagoLast/Exquisite#master","fastly":"^2.2.0","glob":"^7.1.2","http-server":"^0.11.1","jasmine-core":"^2.99.1","jsdoc":"^3.5.5","jsdoc-escape-at":"^1.0.1","karma":"^2.0.0","karma-chrome-launcher":"^2.2.0","karma-jasmine":"^1.1.1","karma-mocha-reporter":"^2.2.5","karma-sourcemap-loader":"^0.3.7","karma-webpack":"^2.0.9","lodash.template":"^4.4.0","mocha":"^5.0.0","puppeteer":"^1.1.0","s3":"^4.4.0","sloc":"^0.2.0","uglifyjs-webpack-plugin":"^1.1.8","webpack":"^3.8.1"},"main":"src/index.js","module":"src/index.js","jsnext:main":"src/index.js","scripts":{"build":"yarn build:dev && yarn build:min","build:dev":"webpack --config webpack/webpack.config.js","build:min":"webpack --config webpack/webpack.min.config.js","build:watch":"webpack -w --config webpack/webpack.config.js","docs":"rm -rf docs/public; jsdoc --configure config/jsdoc/public-conf.json","docs:all":"rm -rf docs/all; jsdoc --configure config/jsdoc/all-conf.json","lint":"eslint .","lint:fix":"eslint . --fix","test":"yarn test:unit && yarn lint && yarn docs","test:unit":"karma start --single-run --browsers ChromeHeadlessNoSandbox test/unit/karma.conf.js","test:watch":"karma start --no-single-run --auto-watch --browsers ChromeHeadlessNoSandbox test/unit/karma.conf.js","test:watchc":"karma start --no-single-run --auto-watch --browsers Chrome test/unit/karma.conf.js","test:user":"karma start --single-run --browsers ChromeHeadlessNoSandbox test/integration/user/karma.conf.js","test:user:watch":"karma start --no-single-run --auto-watch --browsers ChromeHeadlessNoSandbox test/integration/user/karma.conf.js","test:user:watchc":"karma start --no-single-run --browsers Chrome test/integration/user/karma.conf.js","test:browser":"karma start --no-single-run --browsers Chrome test/unit/karma.conf.js","test:render":"yarn build:dev && mocha test/integration/render/render.test.js","test:render:clean":"rm -rf test/integration/render/scenarios/**/**/reference.png","test:render:prepare":"yarn build:dev && node test/integration/render/render.prepare.js ","test:e2e":"yarn build:dev && mocha test/acceptance/e2e.test.js","test:e2e:clean":"rm -rf test/acceptance/e2e/**/reference.png","test:e2e:prepare":"yarn build:dev && node test/acceptance/e2e.prepare.js ","test:benchmark":"node test/benchmark/benchmark.js","serve":"yarn build:dev && yarn docs && http-server","bump":"npm version prerelease","bump:minor":"npm version minor","bump:patch":"npm version patch","release":"./scripts/release.sh","postversion":"git push origin HEAD --follow-tags","ghpublish":"git checkout gh-pages && git pull origin gh-pages && git merge master && yarn build && yarn docs && git commit -a -m \"Auto generated gh-pages\" && git push origin gh-pages && git checkout master","loc":"sloc src/ examples/"}}
```

From the compiled code of this branch. It looks like the full package.json is being included in the code as a node-module.

**Comment by IagoLast** [comment]
We should use [definePlugin](https://webpack.js.org/plugins/define-plugin/)  to define a carto.VERSION property at compile time.



```js
// index.js
export {
    setDefaultAuth,
    setDefaultConfig,
    source,
    expressions,
    Layer,
    Viz,
    Map,
    Interactivity,
    version: VERSION, // eslint-disable-line
};
```

```js
// webpack.config.js
  plugins: [
    new webpack.DefinePlugin({
      VERSION: JSON.stringify(require("package.json").version)
    })
  ]
```

File: src/client/windshaft.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (jgoizueta) [reply src/client/windshaft.js]: It seems the increase represents 3KB of 226 in the minified version, should we care?

**Comment by jgoizueta** [comment]
Finally, we're leaving this as is because we now have tree shaking to remove unused code.

**Pull request merged**